java-strings/src/test/java/com/baeldung/ConvertStringToListUnitTest.java

```diff
@@ -0,0 +1,161 @@
+package com.baeldung;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Test;
+
+import com.google.common.base.Function;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Lists;
+
+public class ConvertStringToListUnitTest {
+
+    private final String countries = "Russia,Germany,England,France,Italy";
+    private final String ranks = "1,2,3,4,5, 6,7";
+    private final String others = ",,,,,";
+    private final List<String> expectedCountriesList = Arrays.asList("Russia", "Germany", "England", "France", "Italy");
+    private final List<Integer> expectedRanksList = Arrays.asList(1, 2, 3, 4, 5, 6, 7);
+    private final List<String> expectedOthersList = Arrays.asList("", "", "", "", "", "");
+
+    @Test
+    public void givenString_thenGetListOfStringByJava() {
+        List<String> convertedList = stringToStringListInJava(countries);

```

Comment:
Sorry if this wasn't clear. What I meant by including your code within the testing file is I think that you can do all of your operations within the test itself. For example, instead of calling `stringToStringListInJava(countries)` you can actually have the code within `stringToStringListInJava` contained within this test. It will make it a lot easier for our readers to follow.